decentraland/unity-renderer
# feat: Decoupling loading screen from kernel

## What does this PR change?

Implements [ADR 101](https://adr.decentraland.org/adr/ADR-101), the decoupling of the loading screen. Behavior should be the same as the current one. Some minor differences should be visible:

1. The only loading message is `Loading scenes, 3D models, and sounds... N% complete`
2. The loading percentage will only show the loading of the destination scene. The change visible here should be a more consistent loading bar.
3. A notification pops-up when there is a loading screen timeout.
4. Solves this [issue](https://app.zenhub.com/workspaces/explorer-6047bad476c3c0001942ee91/issues/decentraland/unity-renderer/1615). No jumpy image should be visible between teleports.

## How to test the changes?

1. Enter DCL through https://play.decentraland.zone/?renderer-branch=feat/decoupling-loading-screen-from-kernel

2. We need to test every teleport method. All of them should take you to the correct destination, with no jumpy image shown before teleporting.

Teleport methods to test are:
Teleport from chat (`/goto coordinates, random, position,magic, crowd`)
Teleport from all the tabs of the Explore panel
Teleport from map
Teleport to any event from the Jump In button present on Genesis Plaza

3. Teleporting to an already loaded scene should not trigger the loading screen. This can be confirmed by teleporting to the current location (Do `/goto 0,0` twice)

4. `/changerealm` command should not trigger the loading screen. Try changing between realms and see that the loading screen is not triggered.

5. Going to a world (and back) should always trigger the loading screen. From (0,0) try going to `/world ile.dcl.eth` and then back to genesis city. The loading screen behaviour should be correct.

6. Same as above, but jump to `/world menduz.dcl.eth` instead of 0,0. Loading screen behaviour should be as expected

7. Test the timeout by going to `/goto 70,70`. There is a scene that always takes more than 30 seconds to load. The loading screen should go away with the notification, but the scene is not going to be loaded

8. Create a new user and check that the signup flow is correctly behaving with the loading screen

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
